### PR TITLE
chore: Removes Saved Query old code

### PR DESCRIPTION
**Pull request by michael-s-molina**
### SUMMARY
This PR removes old code related to the Saved Query module. This code was introduced in https://github.com/apache/superset/pull/7521 and it was later replaced by the new Saved Query module under SQL Lab.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
This is how Saved Queries were displayed back then
Screenshot 2023-08-16 at 14 54 55

This is the new Saved Query module
Screenshot 2023-08-16 at 14 55 22

### TESTING INSTRUCTIONS
CI should be sufficient.

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by dpgaspar — COMMENTED** [review]
can we remove `SavedQueryView` also, or at least of bit chunk of it?

**Comment by michael-s-molina** [comment]
> can we remove `SavedQueryView` also, or at least of bit chunk of it?

No. `SavedQueryView` is used to render the list of saved queries.

**Review by betodealmeida — APPROVED** [review]
Screenshot 2023-08-17 at 09-47-38 chore Removes Saved Query old code by michael-s-molina · Pull Request #25007 · apache_superset

**Review by john-bodley — APPROVED** [review]
Thanks for cleaning up the code base.

**Comment by dpgaspar** [comment]
> > can we remove `SavedQueryView` also, or at least of bit chunk of it?
> 
> No. `SavedQueryView` is used to render the list of saved queries.

right, but just for routing to `savedqueryview/list` then `SavedQueryRestApi` is the one that serves the necessary backend endpoints to render the React List CRUD view, it would nice to limit  `SavedQueryView` exposed endpoints and attributes

**Comment by michael-s-molina** [comment]
> right, but just for routing to savedqueryview/list then SavedQueryRestApi is the one that serves the necessary backend endpoints to render the React List CRUD view, it would nice to limit SavedQueryView exposed endpoints and attributes

@dpgaspar [this commit](https://github.com/apache/superset/pull/25007/commits/4e715d8ec005ede42fec7fafce382e7a15da8c16) cleans up `SavedQueryView`

File: superset/views/sql_lab/views.py [hunk 1]
```diff
@@ -17,120 +17,31 @@
...
 
 
-class SavedQueryView(  # pylint: disable=too-many-ancestors
-    SupersetModelView,
-    DeleteMixin,
-):
-    datamodel = SQLAInterface(SavedQuery)
-    include_route_methods = RouteMethod.CRUD_SET
-
+class SavedQueryView(BaseSupersetView):
```
Comment (dpgaspar) [inline superset/views/sql_lab/views.py:36]: nice!!

**Review by dpgaspar — APPROVED** [review]
Looks great, thank you for doing this!

**Comment by sadpandajoe** [comment]
@michael-s-molina found this [issue](https://github.com/apache/superset/issues/25117) that may be related to your PR.

**Comment by michael-s-molina** [comment]
> @michael-s-molina found this [issue](https://github.com/apache/superset/issues/25117) that may be related to your PR.

Thank you @sadpandajoe. I added a fix [here](https://github.com/apache/superset/pull/25124).

**Pull request merged**